src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -281,10 +294,140 @@ public boolean doesAnomalyDetectorJobIndexExist() {
      *
      * @return true if anomaly result index exists
      */
-    public boolean doesAnomalyResultIndexExist() {
+    public boolean doesDefaultAnomalyResultIndexExist() {
         return clusterService.state().metadata().hasAlias(CommonName.ANOMALY_RESULT_INDEX_ALIAS);
     }
 
+    public boolean doesIndexExist(String indexName) {
+        return clusterService.state().metadata().hasIndex(indexName);
+    }
+
+    public <T> void initCustomResultIndexAndExecute(String resultIndex, AnomalyDetectorFunction function, ActionListener<T> listener) {
```

Comment:
Is it correct to say all validation logic for user to create detector with custom result index is in inside the methods added in AnomalyDetectionIndices class or is there anything additional?

Reply:
And another place is in AnomalyDetector, we will check if result index name match `RESULT_INDEX_NAME_PATTERN`. This is the first release, we don't have BWC issue like detector name.